### ISSUE TITLE
Model improvement

### ISSUE THREAD

**Issue by danielch1** [danielch1/mlops_project#74]
- Model overfits heavily - add some regularization so it achieves higher validation accuracy (>50% - current)
- (optional) - add automatic hyperparameters optimization
